presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -690,7 +690,7 @@ protected Scope visitQuery(Query node, Optional<Scope> scope)
                     .build();
 
             // check column access permissions for each table
-            analysis.getTableColumnReferences()
+            analysis.getTableColumnReferences().get(session.getIdentity())
```

Comment:
Is the `.get(session.getIdentity())` entry guaranteed to exist? 
E.g. `select 1 from view`, `select 1`, `values 1`?

Also, maybe take `Identity` as argument in `Analysis.getTableColumnReferences()`, just as it is in `addTableColumnReferences`.
(If in the future we need a `getTableColumnReferences()` that returns everything, we are free to re-add it)

Reply:
changed it to take an arg and also default to return an empty map meaning that there were no associated tables/column references used.